src/Microsoft.Framework.Configuration/ConfigurationRoot.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.Framework.Configuration
+{
+    public class ConfigurationRoot : IConfigurationRoot
+    {
+        private readonly IList<IConfigurationSource> _sources = new List<IConfigurationSource>();
+
+        public ConfigurationRoot(IList<IConfigurationSource> sources)
+        {
+            _sources = sources;
+        }
+
+        public IEnumerable<IConfigurationSource> Sources
+        {
+            get
+            {
+                return _sources;
+            }
+        }
+
+        public string this[string key]
+        {
+            get
+            {
+                return Get(key);
+            }
+            set
+            {
+                Set(key, value);
+            }
+        }
+
+        public IEnumerable<IConfigurationSection> GetChildren()
+        {
+            return new ConfigurationSection(_sources).GetSection(string.Empty).GetChildren();
+        }
+
+        public IConfigurationSection GetSection(string key)
+        {
+            return new ConfigurationSection(_sources).GetSection(key);
```

Comment:
This is weird as well. Makes me wonder if we should just have a `ConfigurationSection` constructor that takes the sources and the keys and does the right thing.